### PR TITLE
Support search internal index db

### PR DESCRIPTION
**Pull request by leexgh**
Part of: https://github.com/genome-nexus/genome-nexus/issues/540
Search genome nexus index db first, if click "Show more", then search from recoder.

### CODE REVIEW

**Comment by inodb** [comment]
Is there an easy way to enable case insensitive searching?

E.g.
Screen Shot 2021-12-15 at 12 44 22 PM

File: src/component/SearchBox.tsx [hunk 1]
```diff
@@ -143,30 +112,141 @@ export default class SearchBox extends React.Component<ISearchBoxProps> {
...
+                            })
+                            .catch((error: any) => Promise.resolve([]))
+                    );
+                }
+                await Promise.all(promises);
+                if (!this.searchRecoderClicked && !isSearchingByRsid(keyword)) {
+                    enrichedOptions = [
+                        ...enrichedOptions,
+                        {
+                            label: 'Show more searching results',
```
Comment (onursumer) [inline src/component/SearchBox.tsx:229]: let's do
```suggestion
                            label: 'Show more search results',
```
or
```suggestion
                            label: 'Show more results',
```

File: src/component/SearchBox.tsx [hunk 2]
```diff
@@ -143,30 +112,141 @@ export default class SearchBox extends React.Component<ISearchBoxProps> {
...
         }
     };
 
-    @action
-    public setOptions(options: Option[]) {
-        this.options = options;
-    }
+    private getOptions = remoteData<Option[]>({
+        await: () => [],
+        invoke: async () => {
```
Comment (onursumer) [inline src/component/SearchBox.tsx:125]: This function is quite long. Might be better to break it into multiple ones

File: src/util/SearchUtils.ts [hunk 3]
```diff
@@ -38,7 +45,43 @@ export function isValidInput(keyword: string) {
...
+    // input should start with "rs"
+    return keyword.startsWith('rs');
+}
+
+export function isSearchingByHgvsg(keyword: string) {
+    // input should contain ":g."
+    return /:g./i.test(keyword);
+}
+
+export function normalizeInputFormatForInternalDatabaseSearch(keyword: string) {
```
Comment (onursumer) [inline src/util/SearchUtils.ts:58]: Would be nice to have some unit tests for these kind of complex utility functions.

**Review by onursumer — COMMENTED** [review]
Looks good, thanks!

Added just a few minor suggestions & comments.

File: src/util/SearchUtils.ts (hunk 3 above)
Comment (leexgh) [reply src/util/SearchUtils.ts:58]: Currently we don't have any unit tests, but good to add some soon https://github.com/genome-nexus/genome-nexus/issues/573

**Pull request merged**